planet/api/orders.py

```diff
@@ -42,24 +41,16 @@ class OrdersClientException(Exception):
 
 
 class OrdersClient():
-    """High-level access to Planet's orders API.
-
-    Basic Usage::
-
-      from planet.api.orders_client import OrdersClient
-      cl = OrdersClient('api_key')
-      order = cl.get_order('order_id')
+    """High-level asynchronous access to Planet's orders API.
 
-    :param api_key: API key to use. Defaults to environment variable or
-        stored authentication data.
-    :type api_key: str, optional
+    :param session: Open session connected to server
+    :type session: planet.api.http.Session
     :param base_url: The base URL to use. Defaults to production orders API
         base url.
     :type base_url: int, optional
     """
-    def __init__(self, api_key=None, base_url=BASE_URL):
-        api_key = api_key or auth.find_api_key()
-        self.auth = api_key and auth.APIKey(api_key)
+    def __init__(self, session, base_url=BASE_URL):
```

Comment:
For easier usability, it would be nice to make the session arg optional and to instantiate a new session if one isn't provided:
e.g.
```
    def __init__(self, session=None, base_url=BASE_URL, api_key=None):
        if session is None:
            api_key = api_key or somehow_find_api_key()
            session = planet.Session(auth=(api_key, ''))
        self._session = session
```
to allow the user to then do this
```
client = planet.OrdersClient(api_key=some_key)
asyncio.run(client.create_order(order_details))
```

Reply:
@strixcuriosus it seems to me like this approach crosses boundaries between the notions of sessions and clients, which looks like it could be confusing and error-prone. My understanding is that a session is the thing responsible for holding the authentication information and doing stuff, and the client delegates some pre-determined logical flows to the session, assuming that the session knows things about API keys and what-not.

For example:
```python
with planet.Session(auth=planet.get_auth()) as session:
    with planet.OrdersClient(base_url=a_shiny_url) as client:
        oid = asyncio.run(client.create_order(order))
```
There's a bug there, and it wouldn't show up immediately. :-/

I personally _would_ like to see the batteries-included approach with top-level things, like Requests and other libraries do, e.g. `planet.create_order` would create a default `Session` _and_ a default `Client` for you, but not doing the mixing and matching of sessions and clients.

Reply:
+1 to something like a top-level easy to use `planet.create_order` 
My main concern is that this seems harder to use than I think it needs to be.

Reply:
I think a `planet.bulk_create_orders` that leveraged the async client calls under the hood would be great too

Reply:
Thank you @ericrdunham for providing an example that shows how having the client manage the session could be error-prone. If it wasn't for needing a session that served multiple clients (data, orders, etc), the usage proposed by @strixcuriosus would be awesome. I **hear** the need for top-level, usage. I am on board with implementing this. I could use input from both of you on how to make this top-level usage focused and useful. aka, let's come up with some use cases. I have created #256 to capture these use cases. That will be the starting point for implementing top-level helper functions.